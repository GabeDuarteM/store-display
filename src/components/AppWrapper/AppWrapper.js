import React from 'react'
import PropTypes from 'prop-types'
import injectSheet from 'react-jss'
import { Provider } from 'react-redux'

export const styles = {
  [`@global ${
    process.env.NODE_ENV === 'styleguide' ? '*[data-preview]' : 'body'
  }`]: {
    fontFamily: 'Montserrat, sans-serif',
    backgroundColor: '#f5f5f5',
  },
}

const AppWrapper = ({ children, store }) => {
  if (store) {
    return <Provider store={store}>{children}</Provider>
  }
  return children
}

export const propTypes = {
  /** The children that the component will render */
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  /** The redux's store */
  store: PropTypes.object,
}

export const defaultProps = {
  store: null,
}

AppWrapper.propTypes = propTypes
AppWrapper.defaultProps = defaultProps

export default injectSheet(styles)(AppWrapper)
