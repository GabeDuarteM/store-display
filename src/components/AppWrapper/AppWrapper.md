A simple wrapper that injects everything needed for the app to run.

This is shared with Styleguidist, wrapping every component in a AppWrapper, applying the same styles and configs whenever possible

```js
<AppWrapper>
  thing to be <span>wrapped</span>
</AppWrapper>
```
