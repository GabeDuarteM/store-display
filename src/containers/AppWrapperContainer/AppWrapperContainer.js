import React from 'react'

import PropTypes from 'prop-types'

import configureStore from '../../store'
import AppWrapper from '../../components/AppWrapper'

const store = configureStore()

const AppWrapperContainer = ({ children }) => (
  <AppWrapper store={store}>{children}</AppWrapper>
)

export const propTypes = {
  /** The children that is supposed to be rendered */
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
}

AppWrapperContainer.propTypes = propTypes

export default AppWrapperContainer
