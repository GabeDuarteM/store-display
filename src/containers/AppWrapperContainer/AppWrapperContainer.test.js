import React from 'react'

import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import AppWrapperContainer from './AppWrapperContainer'
import AppWrapper from '../../components/AppWrapper'

const createWrapper = (func, children = <div>mock children</div>) =>
  func(<AppWrapperContainer>{children}</AppWrapperContainer>)

const createWrapperShallow = children => createWrapper(shallow, children)

const createWrapperMount = children => createWrapper(mount, children)

describe('COMPONENT: <AppWrapperContainer />', () => {
  it('should render without crashing', () => {
    createWrapperMount()
  })

  it("should match component's snapshot", () => {
    const wrapper = createWrapperShallow()
    expect(toJson(wrapper.dive())).toMatchSnapshot()
  })

  it('should have a AppWrapper component', () => {
    const wrapper = createWrapperShallow()
    const hasAppWrapper = wrapper.find(AppWrapper).exists()

    expect(hasAppWrapper).toBe(true)
  })

  it('should render AppWrapper with the correct props', () => {
    const children = <div>test</div>
    const wrapper = createWrapperShallow(children)
    const hasStoreProp = wrapper.dive().prop('store')
    const childrenProp = wrapper.dive().prop('children')

    expect(hasStoreProp).toBeDefined()
    expect(childrenProp).toBe(children)
  })
})
