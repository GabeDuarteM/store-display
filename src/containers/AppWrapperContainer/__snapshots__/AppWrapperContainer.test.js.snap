// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT: <AppWrapperContainer /> should match component's snapshot 1`] = `
<AppWrapper
  classes={Object {}}
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <div>
    mock children
  </div>
</AppWrapper>
`;
